icc-x-api/icc-helement-x-api.ts

```diff
@@ -267,10 +283,70 @@ export class IccHelementXApi extends IccHelementApi {
       })
   }
 
+  /**
+   * 1. Check whether there is a delegation with 'hcpartyId' or not.
+   * 2. 'fetchHcParty[hcpartyId][1]': is encrypted AES exchange key by RSA public key of him.
+   * 3. Obtain the AES exchange key, by decrypting the previous step value with hcparty private key
+   *      3.1.  KeyPair should be fetch from cache (in jwk)
+   *      3.2.  if it doesn't exist in the cache, it has to be loaded from Browser Local store, and then import it to WebCrypto
+   * 4. Obtain the array of delegations which are delegated to his ID (hcpartyId) in this patient
+   * 5. Decrypt and collect all keys (secretForeignKeys) within delegations of previous step (with obtained AES key of step 4)
+   * 6. Do the REST call to get all helements with (allSecretForeignKeysDelimitedByComa, hcpartyId)
+   *
+   * After these painful steps, you have the helements of the patient.
+   *
+   * @param hcpartyId
+   * @param patient (Promise)
+   * @param keepObsoleteVersions
+   */
+
+  findByUsingPost(hcpartyId: string, patient: models.Patient, keepObsoleteVersions = false) {
```

Comment:
Except the method `findByHCPartyPatientSecretFKeysArray`, the content of this method is exactly the same than findBy: Reuse the code to avoid duplication.

Reply:
Is it ok to have a kind of 'usingPost: boolean = false' parameter to make the switch between the 2 querymethods ?